CMakeLists.txt

```diff
@@ -141,6 +144,18 @@ set(CMAKE_MODULE_PATH
 )
 
 
+# Set linker and compiler flags based on arguments.
+if (VIAMCPPSDK_USE_WALL_WERROR)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror")
+  SET(CMAKE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} -Wall -Werror")
+endif()
+
+if (VIAMCPPSDK_SANITIZED_BUILD)
```

Comment:
@stuqdog - You were having crashes when doing this last time we talked about it. Did you figure out what was going wrong? Do builds with these flags work?

Also, all the same thoughts above about attaching these at the target / directory level. Need to think through whether these should be `PRIVATE` or not.

Reply:
The crashes still exist for purposes of integration testing - as I recall this was the due to gRPC not being built with the sanitizers, so we were running into size issues. I added a comment to this effect. 

Where we stand now is still valuable in the meantime however, as it allows us to still run all unit tests with sanitized builds on. Updating the gRPC build is on the docket along with integration testing support.

Reply:
(update) everything works fine with ubsan and asan on mac. asan is causing issues on linux so I removed it from this PR, will look into it in a subsequent PR.